diagnostic_aggregator/src/aggregator.cpp

```diff
@@ -182,8 +165,17 @@ void Aggregator::diagCallback(const DiagnosticArray::SharedPtr diag_msg)
       if (!analyzed) {
         other_analyzer_->analyze(item);
       }
+
+      // In case there is a degraded state, publish immediately
+      if (critical_ && item->getLevel() > last_top_level_state_) {
```

Comment:
This could lead to problems, because `stale` is larger than everything else.

Reply:
Yes? Would it be bad to publish immediately if items become stale?

Reply:
No, I guess you are right